packages/astro/src/runtime/server/render/script.ts

```diff
@@ -0,0 +1,16 @@
+import type { SSRResult } from '../../../@types/astro.js';
+import { markHTMLString } from '../escape.js';
+
+/**
+ * Relies on the `renderScript: true` compiler option
+ * @experimental
+ */
+export async function renderScript(result: SSRResult, id: string) {
+	const inlined = result.inlinedScripts.get(id);
+	if (inlined) {
+		return markHTMLString(`<script type="module">${inlined}</script>`);
+	}
+
+	const resolved = await result.resolve(id);
+	return markHTMLString(`<script type="module" src="${resolved}"></script>`);
+}
```

Comment:
Would it be possible to avoid duplicates by keeping track of already added scripts on `result`?

Reply:
Oh you're right! I actually wanted to implement this and totally forgot.